### ISSUE TITLE
Bins equals 0 throughs error on histograms

### ISSUE THREAD

**Issue by ciberger** [pandas-profiling/pandas-profiling#884]
**Describe the bug**
Setting plot.histogram.bins to zero causes an error. 

https://github.com/pandas-profiling/pandas-profiling/blob/939969061459a7df5e0ed425cd471827924c4eed/src/pandas_profiling/model/summary_algorithms.py#L29-L45

**To Reproduce**

 

**Version information:**

**Additional context**

**Comment by sbrugman** [comment]
Good catch!  Automated bin detection is (apparently) not supported for weighted data - which we use to greatly improve the performance of the profiling.  This option should be disabled (or implemented).
